File: BeeceptorApiProject/cypress/Integration/Test/CryptocurrencyWallet.js
///<reference types="cypress"/>
import Cryptowallet from "../../support/PageObject/cryptocurrency";
describe('cypto currency wallet api testing', () => {
    let cw=new Cryptowallet();
    let baseurl="https://crypto-wallet-server.mock.beeceptor.com";
    it('Register a user, and create his/her wallet', () => {
        cw.registerUserandCreateWallet('POST', baseurl, '/api/v1/register',{
            "username": "user123",
            "password": "securepassword",
            "email": "user@example.com"
        })
    });
    it('Login the user and generate a session token', () => {
        cw.loginaUserandGenratesessionToken('POST', baseurl, '/api/v1/login',{
            "username": "user123",
            "password": "securepassword"
        })
    });
    it('Retrieve the wallet balance', () => {
       cw.retriveWalletBalance('GET', baseurl, '/api/v1/balance') 
    });
    it('List all the transactions done by the user', () => {
        cw.listTranscationsdoneByuser('GET', baseurl, '/api/v1/transactions')
    });
    it('Transfer 5 ETH to a recipient', () => {
        cw.transfer5ETHtoRecipient('POST', baseurl, '/api/v1/transactions',{
            "recipient_address": "0x1234567890ABCDEF",
            "amount": 5.0,
            "currency": "ETH"
        })
    });
    it('Calculate transaction fees and return estimated cost', () => {
       cw.calculateTransationFeesandReturnEstimatedCost('POST', baseurl, '/api/v1/transaction_fee',{
        "amount": 2.5,
        "currency": "BTC",
        "recipient_address": "0x1234567890ABCDEF"
        
       }) 
    });
    it('Get an object with all available currency exchange rates', () => {
        cw.getObjectwithAllAvailableCurrencyExchangeRates('GET', baseurl, '/api/v1/exchange_rates') 
    });
});

File: ResassuredProject/src/test/java/apiProject/CryptocurrencyWallet.java
package apiProject;
import org.testng.annotations.Test;
import io.restassured.response.Response;
import static io.restassured.RestAssured.*;

import io.restassured.http.ContentType;
import io.restassured.path.json.JsonPath;
import static org.hamcrest.Matchers.*;

import org.json.simple.JSONObject;

public class CryptocurrencyWallet {
	@Test(enabled = true)
	public void Registerauserandcreatewallet() {
		JSONObject js=new JSONObject();
		js.put("username", "Ankita");
		js.put("password", "securepassword");
		js.put("email", "ankita@example.com");
		given().body(js.toJSONString()).when().post("https://crypto-wallet-server.mock.beeceptor.com/api/v1/register").then().statusCode(200).log().all();
	}
	@Test(enabled = true)
	public void Logintheuserandgenerateasessiontoken() {
		JSONObject js=new JSONObject();
		js.put("username", "Ankita");
		js.put("password", "securepassword");
		given().contentType(ContentType.JSON).body(js.toJSONString()).when().post("https://crypto-wallet-server.mock.beeceptor.com/api/v1/login").then().statusCode(200).log().all();
	}
	
	@Test(enabled = true)
	public void Retrievethewalletbalance() {
		given().get("https://crypto-wallet-server.mock.beeceptor.com/api/v1/balance").then().statusCode(200).log().all();
	}
	@Test(enabled = true)
	public void Listallthetransactionsdonebytheuser() {
		given().get("https://crypto-wallet-server.mock.beeceptor.com/api/v1/transactions").then().statusCode(200).log().all();
	}
	@Test(enabled = true)
	public void Transfe5ETHtorecipient() {
		JSONObject js=new JSONObject();
		js.put("recipient_address", "0x1234567890ABCDEF");
		js.put("amount", 5.0);
		js.put("currency", "ETH");
		given().body(js.toJSONString()).when().post("https://crypto-wallet-server.mock.beeceptor.com/api/v1/transactions").then().statusCode(200).log().all();
	}
	
	@Test(enabled = true)
	public void Calculatetransactionfeesandreturnestimatedcost() {
		JSONObject js=new JSONObject();
		js.put("amount", 2.5);
		js.put("currency", "BTC");
		js.put("recipient_address", "0x1234567890ABCDEF");
		given().body(js.toJSONString()).when().post("https://crypto-wallet-server.mock.beeceptor.com/api/v1/transaction_fee").then().statusCode(200).log().all();
	}
	@Test(enabled = true)
	public void Getanobjectwithallavailablecurrencyexchangerates() {
		given().get("https://crypto-wallet-server.mock.beeceptor.com/api/v1/exchange_rates").then().statusCode(200).log().all();
	}
	
}
